test/modules/aws-ecr-integration.ts

```diff
@@ -1,251 +1,494 @@
       WHERE repository_name = '${pubRepositoryName}';
-    `, (res: any[]) => expect(res.length).toBe(0)));
-
-    it('adds a new public repository', query(`
+    `,
+        (res: any[]) => expect(res.length).toBe(0),
+      ),
+    );
+
+    it(
+      'adds a new public repository',
+      query(`
       INSERT INTO public_repository (repository_name)
       VALUES ('${pubRepositoryName}');
-    `));
+    `),
+    );
 
-    it('check adds a new public repository', query(`
+    it(
+      'check adds a new public repository',
+      query(
+        `
       SELECT *
       FROM public_repository
       WHERE repository_name = '${pubRepositoryName}';
-    `, (res: any[]) => expect(res.length).toBe(1)));
+    `,
+        (res: any[]) => expect(res.length).toBe(1),
+      ),
+    );
 
     it('applies the public repository change', apply());
 
-    it('tries to update a public repository autogenerated field', query(`
+    it(
+      'pushes public repository image',
+      query(
+        `SELECT repository_uri FROM public_repository WHERE repository_name='${pubRepositoryName}'`,
+        (res: any[]) => {
+          // get repository uri to retag and pull
+          const repositoryUri = res[0]['repository_uri'];
+          execSync(`docker tag ${dockerImage} ${repositoryUri}`);
+          execSync(
+            `docker login --username AWS -p $(aws ecr-public get-login-password --region ${region}) ${repositoryUri}`,
+          );
+
+          execSync(`docker push ${repositoryUri}`);
+        },
+      ),
+    );
+
+    it('syncs the images', sync());
+
+    it(
+      'check that new images has been created under a public repo',
+      query(
+        `
+      SELECT *
+      FROM repository_image
+      WHERE public_repository = '${pubRepositoryName}';
+    `,
+        (res: any[]) => {
+          expect(res.length).toBe(1);
+        },
+      ),
+    );
+
+    it(
+      'tries to update a public repository autogenerated field',
+      query(`
       UPDATE public_repository SET repository_arn = '${pubRepositoryName}arn' WHERE repository_name = '${pubRepositoryName}';
-    `));
+    `),
+    );
 
     it('applies change which will undo it', apply());
 
-    it('check update public repository (noop)', query(`
+    it(
+      'check update public repository (noop)',
+      query(
+        `
       SELECT *
       FROM public_repository
       WHERE repository_name = '${pubRepositoryName}';
-    `, (res: any[]) => expect(res.length).toBe(1)));
-  
+    `,
+        (res: any[]) => expect(res.length).toBe(1),
+      ),
+    );
+
     it('uninstalls the ecr module', uninstall(modules));
-  
+
     it('installs the ecr module', install(modules));
-  
-    it('deletes the public repository', query(`
+
+    it(
+      'deletes the repository images',
+      query(`
+    DELETE FROM repository_image WHERE public_repository= '${pubRepositoryName}';
+  `),
+    );
+    it('applies deletes the public repository image', apply());
```

Comment:
this should work without this `apply` or this here on purpose? the lazyLoader should figure out the correct order to delete the resources in even if there is a transient error